File: DLL/UnityEngine.CoreModule/UnityEngine/Ray.cs
using System;
using System.Globalization;

namespace UnityEngine;

public struct Ray(Vector3 origin, Vector3 direction) : IFormattable
{
	private Vector3 m_Origin = origin;

	private Vector3 m_Direction = direction.normalized;

	public Vector3 origin
	{
		get
		{
			return m_Origin;
		}
		set
		{
			m_Origin = value;
		}
	}

	public Vector3 direction
	{
		get
		{
			return m_Direction;
		}
		set
		{
			m_Direction = value.normalized;
		}
	}

	public Vector3 GetPoint(float distance)
	{
		return m_Origin + m_Direction * distance;
	}

	public override string ToString()
	{
		return ToString(null, null);
	}

	public string ToString(string format)
	{
		return ToString(format, null);
	}

	public string ToString(string format, IFormatProvider formatProvider)
	{
		if (string.IsNullOrEmpty(format))
		{
			format = "F2";
		}
		if (formatProvider == null)
		{
			formatProvider = CultureInfo.InvariantCulture.NumberFormat;
		}
		return UnityString.Format("Origin: {0}, Dir: {1}", new object[2]
		{
			m_Origin.ToString(format, formatProvider),
			m_Direction.ToString(format, formatProvider)
		});
	}
}

File: Lua/assets/luabuilds/android/arm64/framework/tolua/unityengine/ray.lua
local var_0_0 = rawget
local var_0_1 = setmetatable
local var_0_2 = {
	direction = Vector3.zero,
	origin = Vector3.zero
}
local var_0_3 = tolua.initget({
	direction = Vector3.zero,
	origin = Vector3.zero
})

;({
	direction = Vector3.zero,
	origin = Vector3.zero
}).__index = function(arg_1_0, arg_1_1)
	local var_1_0 = var_0_0(var_0_2, arg_1_1)

	if var_1_0 == nil then
		var_1_0 = var_0_0(var_0_3, arg_1_1)

		if var_1_0 ~= nil then
			return var_1_0(arg_1_0)
		end
	end

	return var_1_0
end
;({
	direction = Vector3.zero,
	origin = Vector3.zero
}).__call = function(arg_2_0, arg_2_1, arg_2_2)
	return var_0_2.New(arg_2_1, arg_2_2)
end
;({
	direction = Vector3.zero,
	origin = Vector3.zero
}).New = function(arg_3_0, arg_3_1)
	local var_3_0 = {
		direction = arg_3_0:Normalize(),
		origin = arg_3_1
	}

	var_0_1(var_3_0, var_0_2)

	return var_3_0
end
;({
	direction = Vector3.zero,
	origin = Vector3.zero
}).GetPoint = function(arg_4_0, arg_4_1)
	(arg_4_0.direction * arg_4_1):Add(arg_4_0.origin)

	return arg_4_0.direction * arg_4_1
end
;({
	direction = Vector3.zero,
	origin = Vector3.zero
}).Get = function(arg_5_0)
	return arg_5_0.origin.x, arg_5_0.origin.y, arg_5_0.origin.z, arg_5_0.direction.x, arg_5_0.direction.y, arg_5_0.direction.z
end
;({
	direction = Vector3.zero,
	origin = Vector3.zero
}).__tostring = function(arg_6_0)
	return string.format("Origin:(%f,%f,%f),Dir:(%f,%f, %f)", arg_6_0.origin.x, arg_6_0.origin.y, arg_6_0.origin.z, arg_6_0.direction.x, arg_6_0.direction.y, arg_6_0.direction.z)
end
UnityEngine.Ray = {
	direction = Vector3.zero,
	origin = Vector3.zero
}

setmetatable({
	direction = Vector3.zero,
	origin = Vector3.zero
}, {
	direction = Vector3.zero,
	origin = Vector3.zero
})

return {
	direction = Vector3.zero,
	origin = Vector3.zero
}
